ignite/metrics/epoch_metric.py

```diff
@@ -68,21 +68,20 @@ def update(self, output: Sequence[torch.Tensor]) -> None:
         if y.ndimension() == 2 and y.shape[1] == 1:
             y = y.squeeze(dim=-1)
 
-        y_pred = y_pred.to(self._predictions)
-        y = y.to(self._targets)
-
-        self._predictions = torch.cat([self._predictions, y_pred], dim=0)
-        self._targets = torch.cat([self._targets, y], dim=0)
+        self._predictions.append(y_pred.detach())
```

Comment:
Note that this changes the behavior.
The old code just cast the Tensors to the right dtype. While your versions does not and adds an extra `.detach()` that is not in the original code.

This is most likely the reason for the CI failures.

Reply:
@albanD thanks for the review !